modules/leveling/cards.py

```diff
@@ -70,43 +70,73 @@ async def generate_rank_card(user, level_data):
-            truncated_name = truncate_text(draw, member.display_name, font_entry, max_name_width)
-            draw.text((110, y + 5), f"#{idx} {truncated_name}", font=font_entry, fill=(255, 255, 255))
-            draw.text((500, y + 5), f"Lvl {level.level}", font=font_entry, fill=(200, 200, 200))
-
-            y += 70
-        except Exception as e:
-            print(f"Error drawing user {level.id}: {e}")
+            discord_user = bot.get_user(level.user) or await bot.fetch_user(level.user)
+        except (NotFound, HTTPException):
+            print(f"Could not fetch user {level.user}. Skipping...")
             continue
 
-    # Export to buffer
-    buffer = BytesIO()
-    image.save(buffer, "PNG")
-    buffer.seek(0)
-
-    return discord.File(buffer, filename="leaderboard.png")
+        display_name = f"➡{discord_user.display_name}" if level.user == current_user_id else discord_user.display_name
+        line = f"{str(idx).zfill(2)} {display_name:<{max_name_len + 4}} Lvl {level.level:<3} | {level.xp} XP"
+        lines.append(line)
+
+    if not lines:
+        lines.append("No users with XP yet.")
+
+    embed.add_field(name="Members", value=f"```{chr(10).join(lines)}```", inline=False)
+
+    embed.set_footer(text=f"Page {page} / {total_pages}")
+
+    return embed
+
+# Keeping this function in as we may want to re-use this in the future
+# async def generate_leaderboard_card(bot, top_users):
+#     width, height = 800, 70 * len(top_users) + 80
+#     image = Image.new("RGBA", (width, height), (13, 13, 13, 255))
+#     draw = ImageDraw.Draw(image)
+#
+#     font_title = ImageFont.truetype(font_path, 28)
+#     font_entry = ImageFont.truetype(font_path, 20)
+#
+#     draw.text((width // 2 - 150, 20), "🏆 Leaderboard", font=font_title, fill=(255, 255, 255))
+#
+#     y = 80
+#     for idx, level in enumerate(top_users, start=1):
+#         if level.xp <= 0:
+#             continue
+#         try:
+#             member = await bot.fetch_user(level.user)
+#             avatar_asset = member.display_avatar.replace(static_format="png")
+#             avatar_bytes = await avatar_asset.read()
+#             avatar = Image.open(BytesIO(avatar_bytes)).resize((50, 50)).convert("RGBA")
+#             avatar = ImageOps.expand(avatar, border=2, fill=(255, 0, 0))
+#             image.paste(avatar, (40, y), avatar)
+#
+#             max_name_width = 250
+#             truncated_name = truncate_text(draw, member.display_name, font_entry, max_name_width)
+#             draw.text((110, y + 5), f"#{idx} {truncated_name}", font=font_entry, fill=(255, 255, 255))
+#             draw.text((500, y + 5), f"Lvl {level.level}", font=font_entry, fill=(200, 200, 200))
+#
+#             y += 70
+#         except Exception as e:
+#             print(f"Error drawing user {level.user}: {e}")
+#             continue
+#
+#     buffer = BytesIO()
+#     image.save(buffer, "PNG")
+#     buffer.seek(0)
+#     return discord.File(buffer, filename="leaderboard.png")
```

Comment:
You shouldn't comment out so much code and track it, you can instead remove it and save the git commit hash to get it back later if you need it, or something